0006-cmx2cdp-protocol/README.md

```diff
@@ -1,84 +1,102 @@
 
-    Solutions:
-    - CDP should broadcast event that new configuration is available.
+- **CMX**: Configuration Management service.
+- **CDP**: Configuration Data Provider service.
+- **CMX2CDP**: name of the protocol used in communication between CMX and any CDP implementation.
 
-2. _Configuration delivery confirmation after configuration push._
+## Requirements and constraints
 
-    Solutions:
-    - CMX should broadcast event that particular endpoint has updated its configuration to specific version.
-    >**NOTE:** In case of CMX configuration pull no delivery confirmation is required since endpoint can
-    recognise delivery fail and request its configuration again.
+CMX2CDP requirements:
 
-3. _It is possible that there are two CDP instances configured to work with different
-content types and one originator service instance. When originator service publishes config event
-message with certain content type it is consumed by both CDPs._
+- Configuration push: broadcasting events about new configuration availability.
+- Configuration delivery confirmation.
+- Filtering messages by content type and communicating them to different CDP instances.
+- Using HTTP status codes and arbitrary reason phrases to inform about configuration unavailability.
 
-    Solutions:
-    - CDP should filter messages by their content types.
+## Use cases
 
-4. _If CMX asks for endpoint configuration but there is no configuration currently available,
-CDP should somehow inform about it._
+### UC1
 
-    Solutions:
-    - Use HTTP status codes and arbitrary reason phrases.
+Once CDP service accepts new configuration, it publishes an event message containing the configuration data.
+The message is published with a [NATS](http://nats.io/) subject, so all services subscribed to this subject will receive the event message.
 
+CDP also listens to configuration-based events and replies to them accordingly.
 
-#### Key concept
+### UC2
+
+Once endpoint is connected, it can send a configuration request to CMX which forwards it to specific CDP.
+
+## Design
 
 There are two messaging approaches used in CMX-CDP communication:
 - [Configuration pull](#configuration-pull)
 - [Configuration push](#configuration-push)
 
 ### Configuration pull
 
-Is used when CMX is intended to request particular configuration.
-
-#### Subject structure
+Configuration pull is used when CMX is intended to request particular configuration.
 
-CMX should send message using NATS to the next subject:
+No delivery confirmation is required for configuration pull, as endpoint can detect delivery fail and request configuration again.
 
-    kaa.v1.service.{cdp-service-instance-name}.cmx2cdp.{message-type}
+#### Subject structure
 
-Also, CMX should include NATS `replyTo` field which points to CMX replica that will handle the response:
+CMX should send messages using this NATS subject:
+```
+kaa.v1.service.{cdp-service-instance-name}.cmx2cdp.{message-type}
```

Comment:
It's for rendering reason -- inline code examples are styled differently from code snippets.
This is inline code example: `kaa.v1.service.{cdp-service-instance-name}.cmx2cdp.{message-type}`
And this is a code snippet:
```
kaa.v1.service.{cdp-service-instance-name}.cmx2cdp.{message-type}
```